tools/REALTEK_RTL8195AM.py

```diff
@@ -0,0 +1,161 @@
+        output.write("".join([chr(long(b, 16)) for b in line]))
+
+def append_image_file(image, output):
+    input = open(image, "rb")
+    output.write(input.read())
+    input.close()
+
+# ----------------------------
+#       main function
+# ----------------------------
+def prepend(image, sym_addr, image_prepend):
+
+    # parse input arguments
+    offset = 44
+
+    output_file = image_prepend
+    output = open(output_file, "wb")
+
+    write_fixed_width_value(os.stat(image).st_size, 8, output)
+    write_fixed_width_value(int(sym_addr), 8, output)
+    write_fixed_width_value(RAM2_RSVD, 16, output)
+
+    append_image_file(image, output)
+    output.close()
+
+def find_section(toolchain, elf, section):
+    ret = None
+    mapfile = elf.rsplit(".", 1)[0] + ".map"
+
+    if toolchain == "GCC_ARM":
+        # .image2.table   0x[00000000]30000000       0x18
+        #                 0x[00000000]30000000       __image2_start__ = .
+        #                 0x[00000000]30000000       __image2_entry_func__ = .
+        with open(mapfile, 'r') as infile:
+            # Search area to parse
+            for line in infile:
+                if line.startswith(section):
+                    match = re.match(r'^' + section + \
+                            r'\s+0x0{,8}(?P<addr>[0-9A-Fa-f]{8})\s+.*$', line)
+                    if match:
+                        ret = match.group("addr")
+                        break
+    elif toolchain in ["ARM_STD", "ARM", "ARM_MICRO"]:
+        # Memory Map of the image
+        #   Load Region LR_DRAM (Base: 0x30000000, Size: 0x00006a74, Max: 0x00200000, ABSOLUTE)
+        #     Execution Region IMAGE2_TABLE (Base: 0x30000000, Size: 0x00000018, Max: 0xffffffff, ABSOLUTE, FIXED)
+        #     Base Addr    Size         Type   Attr      Idx    E Section Name        Object
+        #     0x30000000   0x00000004   Data   RO         5257    .image2.ram.data    rtl8195a_init.o
+        with open(mapfile, 'r') as infile:
+            # Search area to parse
+            for line in infile:
+                if line.startswith('Memory Map of the image'):
+                    break
+            # Search section(block)
+            for line in infile:
+                match = re.match(r'^.*Region\s+' + section + \
+                        r'\s+\(Base: 0x(?P<addr>[0-9A-Fa-f]{8}),\s+Size:.*\)$', line)
+                if match:
+                    ret = match.group("addr")
+                    break
+    elif toolchain == "IAR":
+        #   Section                 Kind        Address     Size  Object
+        #   -------                 ----        -------     ----  ------
+        # "A3":                                           0x8470
+        #   IMAGE2                           0x10006000   0x5d18  <Block>
+        #     .ram_image2.text               0x10006000   0x5bbc  <Block>
+        #       .rodata             const    0x10006000     0x14  retarget.o [17]
+        with open(mapfile, 'r') as infile:
```

Comment:
This line is duplicated in all three of the cases. Could you bring it up one level (in scope) to be the scope above the if statements? that would reduce the amount of code duplication.